manifests/remote_file.pp

```diff
@@ -0,0 +1,20 @@
+# Downloads a file from a URL to a local file given by the title
+define foreman_proxy::remote_file(
+  $remote_location,
+  $mode='0644',
+) {
+  $parent = dirname($title)
+  File <| title == $parent |>
+  -> exec { "mkdir -p ${parent}":
+    path    => ['/bin', '/usr/bin'],
+    creates => $parent,
+  }
+  -> exec {"retrieve_${title}":
+    command => "/usr/bin/curl -s ${remote_location} -o ${title}",
```

Comment:
Is curl definitely going to be installed on all supported platforms?

Reply:
I've updated it to use the plain ```file``` resource for the download. Not sure how well that'll work in practice.